File: java/components/Hardware.java
package components;

import java.util.Date;

public abstract class Hardware extends Component {
    private String serialNumber;
    private Date installationDate;
    private boolean physicallyInstalled;

    public Hardware(String id, String manufacturer, String model, double price, String serialNumber,
            Date installationDate, boolean physicallyInstalled) {
        super(id, manufacturer, model, price);
        this.serialNumber = serialNumber;
        this.installationDate = installationDate;
        this.physicallyInstalled = physicallyInstalled;
    }

    public String getSerialNumber() {
        return serialNumber;
    }

    public void setSerialNumber(String serialNumber) {
        this.serialNumber = serialNumber;
    }

    public Date getInstallationDate() {
        return installationDate;
    }

    public void setInstallationDate(Date installationDate) {
        this.installationDate = installationDate;
    }

    public boolean isPhysicallyInstalled() {
        return physicallyInstalled;
    }

    public void setPhysicallyInstalled(boolean physicallyInstalled) {
        this.physicallyInstalled = physicallyInstalled;
    }
}

File: python/components/hardware.py
from datetime import date

class Hardware:
    def __init__(self, serial_number="", installation_date=None, physically_installed=False):
        self._serial_number = serial_number
        self._installation_date = installation_date if installation_date is not None else date.today()
        self._physically_installed = physically_installed

    # Setters
    def set_serial_number(self, serial_number):
        self._serial_number = serial_number

    def set_installation_date(self, installation_date):
        self._installation_date = installation_date

    def set_physically_installed(self, physically_installed):
        self._physically_installed = physically_installed

    # Getters
    def get_serial_number(self):
        return self._serial_number

    def get_installation_date(self):
        return self._installation_date

    def is_physically_installed(self):
        return self._physically_installed